packages/in_app_purchase/in_app_purchase_storekit/lib/src/in_app_purchase_storekit_platform.dart

```diff
@@ -75,7 +75,10 @@ class InAppPurchaseStoreKitPlatform extends InAppPurchasePlatform {
             purchaseParam is AppStorePurchaseParam ? purchaseParam.quantity : 1,
         applicationUsername: purchaseParam.applicationUserName,
         simulatesAskToBuyInSandbox: purchaseParam is AppStorePurchaseParam &&
-            purchaseParam.simulatesAskToBuyInSandbox));
+            purchaseParam.simulatesAskToBuyInSandbox,
+        paymentDiscount: purchaseParam is AppStorePurchaseParam
```

Comment:
We should test that the `paymentDiscount` is encoded and sent to method channel.

Reply:
I see your point, that in fact the bug was that the discount parameter was not sent trough the method channel. But I am not sure on how to construct such a test?
In my understanding: now the discount is send trough `AppStorePurchaseParam` in the `buyNonConsumable` method, which is not returning a useful result to test. This method constructs `SKPaymentWrapper` with a discount parameter and sends it to `_skPaymentQueueWrapper.addPayment` which turns the `SKPaymentWrapper` to a map (which we test in the newly created test for the toMap method) and sends it trough the method channel. But the channel is not returning useful result for testing either. The encoding of the `paymentDiscount` is in the `toMap` method which we test now. 
Can you elaborate please?

Reply:
You might be able to intercept the method call in the `FakeStoreKitPlatform` and inspect the value?

Reply:
I got your idea. Thanks! Done in the last commit.